digital_asset_types/src/dao/scopes/asset.rs

```diff
@@ -409,3 +413,63 @@ async fn get_grand_total(
     let grand_total = stmt.count(conn).await?;
     Ok(Some(grand_total))
 }
+
+pub async fn add_collection_metadata(
+    conn: &impl ConnectionTrait,
+    mut asset_list: AssetList,
+) -> Result<AssetList, DbErr> {
+    // compile a vec of all the group values (String) from the asset list
+    let mut group_values: Vec<String> = Vec::new();
+    for item in &asset_list.items {
+        if let Some(groups) = item.grouping.clone() {
+            for group in groups {
+                let group_value = group.group_value;
+                if let Some(group_value) = group_value {
+                    group_values.push(group_value);
+                }
+            }
+        }
+    }
+
+    // convert the group values to bytea (bc asset_data.id is bytea)
+    let bytea_group_values: Vec<Vec<u8>> = group_values
+        .iter()
+        .map(|group_value| {
+            let bytea_group_value = group_value.as_bytes().to_vec();
+            bytea_group_value
+        })
+        .collect();
+
+    // make a query to fetch all the metadata
+    let asset_data = asset_data::Entity::find()
+        .filter(asset_data::Column::Id.is_in(bytea_group_values))
+        .all(conn)
+        .await?;
+
+    // create a mapping of id -> collection_metadata
+    let mut hashmap: HashMap<String, CollectionMetadata> = HashMap::new();
+    for data in &asset_data {
+        let id = String::from_utf8_lossy(&data.id).to_string();
+        let (name, symbol) = process_raw_fields(&data.raw_name, &data.raw_symbol);
```

Comment:
use regular name/symbol not raw. raw is very specific for powerusers like tensor and the data isnt fully populated